core/src/main/java/org/apache/accumulo/core/spi/compaction/CompactionsDirectiveImpl.java

```diff
@@ -18,64 +18,28 @@
  */
 package org.apache.accumulo.core.spi.compaction;
 
-import java.util.Objects;
-
-import org.apache.accumulo.core.spi.compaction.CompactionDirectives.Builder;
-
-import com.google.common.base.Preconditions;
-
 /**
- * This class intentionally package private. This implementation is odd because it supports zero
- * object allocations for {@code CompactionDirectives.builder().build()}.
+ * This class intentionally package private. It supports one object allocation for
+ * {@code CompactionDirectives}.
```

Comment:
```suggestion
 * This class intentionally package private.
```

The old code implemented a copy on write strategy to avoid object allocation when building the default.  If that is to be removed I don't think a comment is needed saying objects are allocated.

Reply:
I am going to rename this class to `CompactionDirectivesImpl`. This follows the interface name. It was also the only class with an 's' on Compaction for some reason, possibly a typo.